tests/unit/basic_crawler/test_basic_crawler.py

```diff
@@ -796,31 +796,16 @@ async def handler(context: BasicCrawlingContext) -> None:
     }
 
 
-async def test_passes_configuration_to_storages() -> None:
-    configuration = Configuration(persist_storage=False, purge_on_start=True)
-
-    crawler = BasicCrawler(configuration=configuration)
-
-    dataset = await crawler.get_dataset()
-    assert dataset._configuration is configuration
-
-    key_value_store = await crawler.get_key_value_store()
-    assert key_value_store._configuration is configuration
-
-    request_provider = await crawler.get_request_provider()
-    assert isinstance(request_provider, RequestQueue)
-    assert request_provider._configuration is configuration
-
-
 async def test_respects_no_persist_storage() -> None:
-    configuration = Configuration(persist_storage=False)
-    crawler = BasicCrawler(configuration=configuration)
+    config = Configuration(persist_storage=False)
+    service_locator.set_configuration(config)
+    crawler = BasicCrawler()
```

Comment:
I think the previous usage was fine and also more in line with what a user might do.